File: apps/android/src/main/java/tv/every/tilawah/android/telemetry/NoOpTelemetry.kt
package tv.every.tilawah.android.telemetry

import tv.every.tilawah.android.app.AppError

/**
 * Discards every telemetry call. Used by Compose previews and unit
 * tests where Logcat noise would obscure the signal.
 */
object NoOpTelemetry : Telemetry {
    override fun event(name: String, attributes: Map<String, String>) = Unit

    override suspend fun <T> measure(name: String, block: suspend () -> T): T = block()

    override fun error(error: AppError, context: Map<String, String>) = Unit
}

File: apps/ios/Sources/QuranRecitationApp/Telemetry/NoOpTelemetry.swift
import Foundation

/// Telemetry that drops every signal. Used by SwiftUI Previews and as
/// the default for tests that do not assert on telemetry. When tests
/// need to assert, PR 6 introduces `TelemetrySpy` in the test target.
struct NoOpTelemetry: Telemetry {
  func event(_ name: String, attributes: [String: String]) {}

  func measure<T>(_ name: String, _ block: () async throws -> T) async rethrows -> T {
    try await block()
  }

  func error(_ error: AppError, context: [String: String]) {}
}
